streamalert/alert_processor/outputs/teams.py

```diff
@@ -0,0 +1,217 @@
+            )
+
+        return teams_card
+
+    @staticmethod
+    def _add_additional_sections(teams_card, additional_sections):
+        """Add additional card sections to the teams card
+
+        Args:
+            teams_card (pymsteams.connectorcard): Teams connector card
+            additional_sections (list[pymsteams.cardsection]):
+                Additional sections to be added to the card. Each section should be of
+                type: pymsteams.cardsection and have their relevant fields filled out.
+                Please review the pymsteams documentation for additional information.
+
+        Returns:
+            teams_card (pymsteams.connectorcard): teams_card with additional sections added
+        """
+        if not isinstance(additional_sections, list):
+            LOGGER.debug("additional_sections is not a list, converting")
+
+            additional_sections = [additional_sections]
+
+        for additional_section in additional_sections:
+            if not isinstance(additional_section, pymsteams.cardsection):
+                LOGGER.error(
+                    "additional_section: %s is not an instance of %s",
+                    additional_section,
+                    pymsteams.cardsection,
+                )
+                continue
+
+            teams_card.addSection(additional_section)
+
+        return teams_card
+
+    def _dispatch(self, alert, descriptor):
+        """Sends the Teams Card to Teams
+
+        Publishing:
+            By default the teams output sends a teams card comprising some default intro text
+            and a section containing:
+            * title with rule name
+            * alert description
+            * alert record (as a section of key/value pairs)
+
+            To override this behavior use the following fields:
+
+            - @teams.title (str):
+                Replaces the title of the teams connector card.
+
+            - @teams.description (str):
+                Replaces the text of the team connector card
+
+            - @teams.card_color (str):
+                Replaces the default color of the connector card (red)
+                Note: colors are represented by hex string
+
+            - @teams.with_record (bool):
+                Set to False, to remove the alert record section. Useful if you want to have a
+                more targeted approach for the alert
+
+            - @teams.additional_card_sections (list[pymsteams.cardsection]):
+                Pass in additional sections you want to send on the message.
+
+                @see cls._add_additional_sections() for more info
+
+        Args:
+            alert (Alert): Alert instance which triggered a rule
+            descriptor (str): Output descriptor
+
+        Returns:
+            bool: True if alert was sent successfully, False otherwise
+        """
+        creds = self._load_creds(descriptor)
+        if not creds:
```

Comment:
I would log an error message if no creds provided.

Reply:
I copied the styling from: https://github.com/airbnb/streamalert/blob/b87f7769c1e93f74f7072d92da1961012f928ecb/streamalert/alert_processor/outputs/slack.py#L446

As this returns `True`/`False`